### PR TITLE
Fix upstream response and add size check

### PR DESCRIPTION
**Pull request by vicancy**
Fix https://github.com/Azure/azure-signalr/issues/1152

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Emulator/HubEmulator/HttpServerlessMessageHandler.cs [hunk 1]
```diff
@@ -60,7 +60,37 @@ public async Task<ReadOnlySequence<byte>> WriteMessageAsync(HubConnectionContext
...
+                        {
+                            // same as no content.
+                            await connectionContext.WriteAsync(CompletionMessage.Empty(message.InvocationId));
+                        }
+                        else if (contentLength > 16 * 1024 * 1024)
+                        {
+                            // We don't support response large than 16M, fast fail.
+                            await connectionContext.WriteAsync(CompletionMessage.WithError(message.InvocationId, $"Invocation failed, response too large."));
+                        }
+                        else
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Emulator/HubEmulator/HttpServerlessMessageHandler.cs:74]: what about `null`?

Comment (vicancy) [reply src/Microsoft.Azure.SignalR.Emulator/HubEmulator/HttpServerlessMessageHandler.cs:74]: which is null?

Comment (vicancy) [reply src/Microsoft.Azure.SignalR.Emulator/HubEmulator/HttpServerlessMessageHandler.cs:75]: Added LimitedStream

File: src/Microsoft.Azure.SignalR.Emulator/HubEmulator/HttpServerlessMessageHandler.cs [hunk 2]
```diff
@@ -60,7 +61,38 @@ public async Task<ReadOnlySequence<byte>> WriteMessageAsync(HubConnectionContext
...
                     if (response.StatusCode == HttpStatusCode.OK)
                     {
-                        return message.Payload;
+                        var contentLength = response.Content.Headers.ContentLength;
+                        if (contentLength == 0)
+                        {
+                            // same as no content.
+                            await connectionContext.WriteAsync(CompletionMessage.Empty(message.InvocationId));
+                        }
+                        else if (contentLength > 16 * 1024 * 1024)
```
Comment (JialinXin) [inline src/Microsoft.Azure.SignalR.Emulator/HubEmulator/HttpServerlessMessageHandler.cs:70]: >16 * 1024 * 1024 [](start = 49, length = 16)

make this max length constants?

**Review by JialinXin — APPROVED** [review]
:shipit:

**Pull request merged**